MonoDevelop.MSBuild.Editor/Completion/MSBuildCompletionSource.cs

```diff
@@ -119,19 +125,141 @@ CompletionContext CreateCompletionContext (List<CompletionItem> items)
 
 			foreach (var att in rr.GetAttributeCompletions (doc, doc.ToolsVersion)) {
 				if (!existingAtts.ContainsKey (att.Name)) {
-					items.Add (CreateCompletionItem (att));
+					items.Add (CreateCompletionItem (att, MSBuildSpecialCommitKind.Attribute));
 				}
 			}
 
 			return CreateCompletionContext (items);
 		}
 
-		CompletionItem CreateCompletionItem (BaseInfo info, string prefix = null)
+		//Dictionary<int, string> cache = new Dictionary<int, string> ();
+
+		protected override async Task<CompletionContext> GetAttributeValueCompletionsAsync (
+			IAsyncCompletionSession session,
+			SnapshotPoint triggerLocation,
+			List<XObject> nodePath,
+			IAttributedXObject attributedObject,
+			XAttribute attribute,
+			CancellationToken token)
+		{
+			/*if (!attribute.NameEquals ("include", true)) {
+				return CompletionContext.Empty;
+			}*/
+
+			var context = await GetSessionContext (session, triggerLocation, token);
+			var rr = context.rr;
+			var doc = context.doc;
+
+			if (rr == null) {
+				return CompletionContext.Empty;
+			}
+
+			if (attributedObject is XElement xElement &&
+				xElement.NameEquals ("PackageReference", true)) {
+
+				switch (Regex.Replace (attribute.FriendlyPathRepresentation.ToLower (), @"[^0-9a-zA-Z]+", "")) {
+				case "include":
+					return await GetPackageNamesAsync (doc, rr);
+
+				case "version":
+					return await GetPackageVersionCompletions (doc, rr, token);//GetPackageVersionsAsync (doc, rr);
+				}
+			}
+
+			return CompletionContext.Empty;
+		}
+
+		async Task<CompletionContext> GetPackageNamesAsync(MSBuildRootDocument doc,
+														MSBuildResolveResult rr)
+        {
+			if (rr == null) {
+				return CompletionContext.Empty;
+			}
+
+			var items = new List<CompletionItem> ();
+			var httpClient = new HttpClient ();
+			string result = null;
+			var searchQuery = rr.Reference.ToString ().ToLower ();
+			if (string.IsNullOrEmpty (searchQuery)) {
+				return CompletionContext.Empty;
+			}
+
+			try {
+				var cancellationPackage = new CancellationTokenSource (TimeSpan.FromSeconds (10));
+				var packageResponse = await httpClient.GetAsync ($"https://api-v2v3search-0.nuget.org/autocomplete?q={searchQuery}&skip=0&take=100", cancellationPackage.Token);
+				result = await packageResponse.Content.ReadAsStringAsync ();
+
+			} catch (Exception) {
+				return CompletionContext.Empty;
+			}
+
+			var packageJSONObject = JObject.Parse (result);
+			var packageArray = (JArray)packageJSONObject["data"];
+
+			foreach (var package in packageArray) {
+				var info = package.ToString ();
+				items.Add (CreateCompletionItem (new ItemInfo (info, null, null),
+						MSBuildSpecialCommitKind.AttributeValueSpecial));
+			}
+			return new CompletionContext (ImmutableArray<CompletionItem>.Empty.AddRange (items));
+		}
+
+		async Task<CompletionContext> GetPackageVersionsAsync (MSBuildRootDocument doc,
```

Comment:
This is redundant, `GetPackageVersionCompletions` already handles this.

Reply:
Oh, just saw your comment

> Package Version Search - GetPackageVersionsAsync is not been used anymore. Letting it be there for future purposes. Instead, GetPackageVersionCompletions been used now.

What future purposes do you expect this might have? It would be good to have a comment explaining them so folks reading the file understand why it's there.

Reply:
I have already removed it in my most recent commit.